lib/rom/sql/attribute.rb

```diff
@@ -235,6 +257,11 @@ def sql_expr
         @sql_expr ||= (meta[:sql_expr] || Sequel[to_sym])
       end
 
+      # @api private
+      def type_extensions
+        @type_extensions ||= TypeExtensions[unwrap_type(type)] || EMPTY_HASH
+      end
```

Comment:
I know we need to, but we can avoid doing that at run-time and we can keep Attribute API smaller (just an option with reader, instead of 2 methods and lazy assignment). Do you think it makes sense to extend Attribute with Options, or just add a second optional arg to its constructor? We may need options in attributes but it could slow things down, as we create a lot of new attribute objects at run-time.